localstack/services/firehose/provider.py

```diff
@@ -779,3 +795,76 @@ def _get_s3_object_path(self, stream_name, prefix):
         path = pattern.format(pre=prefix, name=stream_name, rand=str(uuid.uuid4()))
         path = timestamp(format=path)
         return path
+
+    def _put_to_redshift(
+        self,
+        stream_name: str,
+        records: List[Dict],
+        redshift_destination_description: RedshiftDestinationDescription,
+    ):
+        jdbcurl = redshift_destination_description.get("ClusterJDBCURL")
+        cluster_id = self._get_cluster_id_from_jdbc_url(jdbcurl)
+        db_name = jdbcurl.split("/")[-1]
+        table_name = redshift_destination_description.get("CopyCommand").get("DataTableName")
+
+        columns_str = None
+        rows_to_insert = []
+        for record in records:
+            data = base64.b64decode(record.get("Data") or record.get("data"))
+            data = to_str(data)
+            data = json.loads(data)
+
+            if columns_str is None:
+                columns_str = ", ".join(data.keys())
+
+            values = []
+            for value in data.values():
+                if isinstance(value, str):
+                    # Escaping single quotes by doubling them
+                    value = value.replace("\t", " ")
+                    value = f"'{value}'"
+                elif value is None:
+                    value = "NULL"
+                else:
+                    value = str(value)
+                values.append(value)
+
+            value_str = ",".join(values)
+            rows_to_insert.append(value_str)
+
+        sql_insert_statement = f"INSERT INTO {table_name} ({columns_str}) VALUES "
+        for row in rows_to_insert:
+            sql_insert_statement += f"({row}),"
+        sql_insert_statement = sql_insert_statement.rstrip(",") + ";"
+
+        execute_statement = {
+            "Sql": sql_insert_statement,
+            "Database": db_name,
+            "ClusterIdentifier": cluster_id,  # cluster_identifier in cluster create
+        }
+
+        if role_arn := redshift_destination_description.get("RoleARN"):
+            factory = connect_to.with_assumed_role(
+                role_arn=role_arn, service_principal=ServicePrincipal.firehose
+            )
+        else:
+            factory = connect_to()
```

Comment:
In this situation we need to get the account id and secret from the request context. See other usages of `connect_to`, e.g.

https://github.com/localstack/localstack/blob/76292e1d01d3141cbedfc0aea9f7b36095d6a23a/localstack/services/kinesis/provider.py#L92-L94

Reply:
fixed in: https://github.com/localstack/localstack/commit/3b33957d536cc34f5244be94ced08fb9556c79ce